src/main/java/se/iths/crimedatabase/controller/AddressController.java

```diff
@@ -19,36 +22,51 @@ public AddressController(AddressService addressService) {
 
     @PostMapping
     public ResponseEntity<Address> create(@RequestBody Address address) {
+        if (address.getStreetAddress().isEmpty() || address.getCity().isEmpty() || address.getZipCode().isEmpty())
+            throw new BadRequestException("Street address, city and zipcode cannot be empty");
+
         Address createdAddress = addressService.create(address);
         return new ResponseEntity<>(createdAddress, HttpStatus.CREATED);
     }
 
     @DeleteMapping("{id}")
     public ResponseEntity<Void> delete(@PathVariable Long id) {
+        if (addressService.findById(id).isEmpty())
```

Comment:
This does not work in Spring, it will return a 500 Internal Server Errror
![image](https://user-images.githubusercontent.com/89127725/159258778-4fc9ae5d-a05c-45f5-8a7c-0ef7a2592f2f.png)

Reply:
Ok!